packages/web-components/src/components/rux-pop-up-menu/rux-pop-up-menu.tsx

```diff
@@ -1,276 +1,300 @@
+        computePosition(this.triggerSlot, this.content, {
+            placement: this.placement,
+            middleware: [offset(12), flip(), arrow({ element: this.arrowEl })],
+        }).then(({ x, y, placement, middlewareData }) => {
+            Object.assign(this.content.style, {
+                left: `${x}px`,
+                top: `${y}px`,
+            })
+
+            const { x: arrowX, y: arrowY } = middlewareData.arrow
+
+            const staticSide = {
+                top: 'bottom',
+                right: 'left',
+                bottom: 'top',
+                left: 'right',
+            }[placement.split('-')[0]]
+
+            Object.assign(this.arrowEl.style, {
+                left: arrowX != null ? `${arrowX}px` : '',
+                top: arrowY != null ? `${arrowY}px` : '',
+                right: '',
+                bottom: '',
+                [staticSide]: '-6px',
+            })
+        })
+        this._determineArrowPosition()
     }
 
-    private _bindElements() {
-        // find and set triggerEl from aria-controls if not given
-        if (!this.triggerEl) {
-            const triggerEl: HTMLElement | null = document.querySelector(
-                `[aria-controls="${this.el.id}"]`
+    private _startPositioner() {
+        this._stopPositioner()
+        if (this.open) {
+            this._position()
+            this._positionerCleanup = autoUpdate(
+                this.triggerSlot,
+                this.content,
+                this._position.bind(this)
             )
-
-            if (triggerEl) {
-                this.triggerEl = triggerEl
-                this.triggerEl.addEventListener('mousedown', this._handleClick)
-            }
-        } else {
-            this.triggerEl.addEventListener('mousedown', this._handleClick)
-        }
-
-        // If a trigger element exists but no anchor, assign trigger to anchor
-        if (!this.anchorEl && this.triggerEl) {
-            this.anchorEl = this.triggerEl
-            this.anchorBounds = this.anchorEl.getBoundingClientRect()
-        } else if (this.anchorEl) {
-            this.anchorBounds = this.anchorEl.getBoundingClientRect()
         }
-        this.menuBounds = this.el.getBoundingClientRect()
     }
 
-    private _setMenuPosition() {
-        if (this.anchorEl && this.anchorBounds && this.menuBounds) {
-            let { anchorBounds, menuBounds } = this
-            anchorBounds = this.anchorEl.getBoundingClientRect()
-            menuBounds = this.el.getBoundingClientRect()
-            const caret = parseInt(getComputedStyle(this.el, ':after').height)
-            let top: number
-            let left: number
-            let caretLeft: number
-
-            const padding = 8
+    /**
+     * This returns which side the arrow is on: top, right, left or bottom.
+     * Currently using this to determine which border to bolster, but could be useful in the future.
+     * ! This does cause a 'state changed during render' stencil erorr.
```

Comment:
why is arrowPosition using @state? can we just make it a private var?

Reply:
I'm using state to trigger a re-render in order to get the correct styles applied to it. It would work as a private var if the pop-up never swapped where the arrow is, but since we're using auto-placement I need the re-render. There might be a better solution though, open to ideas